### PR TITLE
Enable client plugin

### PR DESCRIPTION
**Pull request by czhen-zilliz**

### CODE REVIEW

File: client/config-overrides.js [hunk 1]
```diff
@@ -1,12 +1,7 @@
...
-module.exports = {
-  // The paths config to use when compiling your react app
-  //  for development or production.
-  paths: function (paths, env) {
-    // ...add your paths config
-    paths.appBuild = path.resolve(__dirname, '../server/build');
-    return paths;
-  },
-};
+const aliasMap = configPaths('./tsconfig.paths.json'); // or jsconfig.paths.json
```
Comment (nameczz) [inline client/config-overrides.js:5]: what's the comments means?

File: client/src/router/Config.ts [hunk 2]
```diff
@@ -32,11 +34,30 @@ const RouterConfig: RouterConfigType[] = [
...
 ];
 
+function importAll(r: any, outOfRoot = false) {
+  r.keys().forEach((key: any) => {
+    const content = r(key);
+    const dirName = key.split('/config.json').shift().split('/')[1];
+    const pathName = content.client?.path;
+    const entry = content.client?.entry;
+    if (!(pathName || entry)) return;
+    console.log(content);
```
Comment (nameczz) [inline client/src/router/Config.ts:46]: clean console

File: client/src/router/Config.ts [hunk 3]
```diff
@@ -32,11 +34,30 @@ const RouterConfig: RouterConfigType[] = [
...
-  },
 ];
 
+function importAll(r: any, outOfRoot = false) {
+  r.keys().forEach((key: any) => {
+    const content = r(key);
+    const dirName = key.split('/config.json').shift().split('/')[1];
+    const pathName = content.client?.path;
+    const entry = content.client?.entry;
+    if (!(pathName || entry)) return;
```
Comment (nameczz) [inline client/src/router/Config.ts:45]: it's looks like we both need pathName and entry ?

File: client/tsconfig.paths.plugin.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "compilerOptions": {
```
Comment (nameczz) [inline client/tsconfig.paths.plugin.json:2]: How to use this file?

**Review by nameczz — CHANGES_REQUESTED** [review]

File: client/src/router/Config.ts (hunk 3 above)
Comment (czhen-zilliz) [reply client/src/router/Config.ts]: Has renamed `entry` to `fileEntry`.
`pathName` for route path.
`fileEntry` for load component from target file

File: client/tsconfig.paths.plugin.json (hunk 4 above)
Comment (czhen-zilliz) [reply client/tsconfig.paths.plugin.json:2]: For outside plugin dev mode using.

**Review by nameczz — APPROVED** [review]

**Pull request merged**